### ISSUE TITLE
Can this tool actually crawl/spider, or just scrape pages?

### ISSUE THREAD

**Issue by bogdancss** [bda-research/node-crawler#383]
Hey,

I may not fully understand these terms, but can this tool actually crawl/spider all the pages under a domain, or does it just scrape a specific url?

When I say crawl/spider, I am thinking of something like the ScreamingFrog Spider tool, where you can provide an url, and it will find all (most) other pages on that site.

Please feel free to close this issue, but I feel the tool description needs to be a bit more clear.

Thanks

**Comment by paulkre** [comment]
I agree, `node-scraper` would be a more fitting name for this tool. Or is there an easy configuration to make it behave like an actual crawler?

**Comment by mike442144** [comment]
Yes, "scraper" should be much better. But never mind, you may implement a spider by yourself based on this.

**Comment by raquelmsmith** [comment]
> But never mind, you may implement a spider by yourself based on this.

how do you do this?

**Comment by mike442144** [comment]
1. Figure out the home page or entrance URL which is good to start;
2. Send request to the URL(s);
3. Parse the page content that you get from the response to get all the URLs you care about, which may be the same domain as the previous one;
4. Save the page content to a file or Db whatever you want;
5. Repeat from step 2 to end.

**Comment by franciscohanna92** [comment]
Here is a simple implementation of a spider that can crawl a list of advanced search results from IMDB and extract the title of each movie in the search results

```javascript
import Crawler from "crawler";

const BASE_URL = "https://www.imdb.com";
const START_URL = `${BASE_URL}/search/title/?title_type=feature&groups=top_1000&countries=us`;

const pagesCrawler = new Crawler({
  maxConnections: 10,
  callback: (error: any, res: any, done: any) => {
    if (error) {
      console.log(error);
    } else {
      const $ = res.$;

      // Find each title link and queue it in the movieCrawler
      $(".lister")
        .find(".lister-item-header > a")
        .each((i: number, elem: any) => {
          movieCrawler.queue(`${BASE_URL}${elem.attribs.href}`);
        });

      // Find the next page link and queue it in the pagesCrawler
      const nextPage = $(".next-page").attr("href");
      if (nextPage) {
        pagesCrawler.queue(`${BASE_URL}${nextPage}`);
      }
    }
    done();
  },
});

const movieCrawler = new Crawler({
  maxConnections: 10,
  callback: (error: any, res: any, done: any) => {
    if (error) {
      console.log(error);
    } else {
      const $ = res.$;

      // Extract the movie info
      console.log($("title").text());
    }
    done();
  },
});

// Start the crawler
pagesCrawler.queue([START_URL]);

```

**Comment by mike442144** [comment]
solved in #420
